File: Ashoka Uni Year 3/Sem 1/PLDI/assignment 2/Code/Python/main.py
input_arr= input().split()

num_arr=[]

for i in range(len(input_arr)):
  num_arr.append(int(input_arr[i]))

main_array=[]
counter=0

for i in range(0,len(num_arr)):

  temp_arr=[]
  temp_arr.append(int(num_arr[i]))
  counter=1

  for j in range(i+1,len(num_arr)):

    if(num_arr[j]>temp_arr[counter-1]):

      temp_arr.append(num_arr[j])
      counter=counter+1

  if(counter>len(main_array)):

    main_array=temp_arr
    
print (main_array)

File: Ashoka Uni Year 3/Sem 1/PLDI/assignment 2/Code/Rust/main.rs
use std::io;

fn main() {
  let mut input = String::new();

  io::stdin().read_line(&mut input)
      .expect("Couldn't read line");
  let temp_input=input.trim();
  let input_arr = temp_input.split(" ");

  let mut main_arr = vec![0];
  main_arr.pop();

  let mut max_count=0;
  let mut iter_count=0;
  let mut input_vec = vec![0];
  input_vec.pop();
  
  for i in input_arr{
    let myint: i32 = i.parse().unwrap();
    input_vec.push(myint);
    }
    
  

  for i in 0..(input_vec.len()){
    let mut hold_arr= vec![input_vec[i]];
    let mut count=1;

    for j in iter_count..(input_vec.len()){
      if input_vec[j]>hold_arr[count-1]{
        hold_arr.push(input_vec[j]);
        count=count+1;
      }

    }

    if count>max_count{
      max_count=count;
      for k in 0..(main_arr.len()){
        main_arr.pop();
      
      }
      for k in 0..(count){
        main_arr.push(hold_arr[k])
      }
    }
    iter_count=iter_count+1;
  }
  for i in main_arr.iter_mut(){
    print!("{} ",i);
  }
  
}